site/docs/features/index.md

```diff
@@ -1,107 +1,327 @@
+into the "master"-branch.
 ```
+          task#1         task#2   "work"
+          result         result   branch
+            |                |     |
+            v                v     v
+      +-----------+       +-----------+
+      | commit #2 | --<-- | commit #3 |
+      +-----------+       +-----------+
+         |                          |        
+         v                          ^
+         |                          |
++-----------+                     +-----------+
+| commit #1 | --------<---------- | commit #4 |  
++-----------+                     +-----------+
+                                      ^
+                                      |
+                                      |
+                                   "master"
+                                    branch
+```
+Note that the merge-`commit #4` has references to two commits:
+
+* the previous commit on the "master" branch and
+* the merged commit from the "work"-branch
+
+It would be nice to give that `commit #4` a meaningful commit message, like
+`aggregate-financial-stuff 2020/12/24`, though, so people that look through the
+history of the data can grasp what that commit changes and why it's there.
+
+Before you actually merge the data from a working-branch into the "master" branch,
+it is possible to review all the changes made by your job - either programmatically,
+like running some process that validates the state of the branch, or asking a human
+to take a look.
+
+### Working branches for "humans"
+
+You can also use "developer" branches to run experiments against your data, test
+changes of your jobs, etc etc etc.
+
+Production and staging and development can use the same data lake without risking
+the consistent state of production data.
+
+### Squashing commits
+
+Nessie can also help to reduce the amount of data files in a data lake.
+
+You can for example merge all data files produces by all tasks of an analytics job
+before the whole change is actually merged into the "master" branch. Nessie will
+[figure out](#garbage-collection) that only the merged files need to survive and
+remove the unused files.
+
+### Tags
+
+Another type of named references are *tags*. Nessie *tags* are named references
+to specific commits but unlike branches do not get automatically "moved".
+
+*Tags* are useful to reference specific commits, for example a tag named
+`financial-data-of-FY2021` could reference all sources of financial data relevant
+used for some FY report.
+
+### Commit messages and more
+
+As briefly mentioned above, every commit in Nessie has a bunch of attributes.
+The probably most important ones are "summary" and "description", which are exactly
+that - meaningful summaries and detailed descriptions that explain what has been
+changed and why it has been changed. This gives data engineers the ability to know
+why something has changed.
 
-## Data and Metadata
+In addition to "summary" and "description", there are a bunch of additional attributes:
```

Comment:
Our commit message support is a bit squishy in Nessie currently. The plan is to add mroe structure but right now its just a freehand msg, timestamp and author. Could we skip this section till its better defined?

Reply:
I've added a sentence explaining that we intend to add more structure to these attributes, so people know that these attributes exist but will get some more structure, given that most commits will be machine-generated ones. WDYT?